iree_tests/benchmarks/sdxl/benchmark_sdxl_rocm.py

```diff
@@ -159,46 +163,145 @@ def job_summary_process(ret_value, output):
     return benchmark_mean_time
 
 def test_sdxl_rocm_benchmark(goldentime_rocm_e2e, goldentime_rocm_unet, 
-    goldentime_rocm_clip, goldentime_rocm_vae, gpu_number, rocm_chip):
-    # if the benchmark returns 1, benchmark failed
-    job_summary_lines = []
+    goldentime_rocm_clip, goldentime_rocm_vae, gpu_number, rocm_chip,
+    goldendispatch_rocm_unet, goldendispatch_rocm_clip, goldendispatch_rocm_vae,
+    goldensize_rocm_unet, goldensize_rocm_clip, goldensize_rocm_vae):
 
     # e2e benchmark
     ret_value, output = run_sdxl_rocm_benchmark(rocm_chip, gpu_number)
     benchmark_mean_time = job_summary_process(ret_value, output)
-    assert benchmark_mean_time <= goldentime_rocm_e2e, "SDXL e2e benchmark time should not regress"
+    check.less_equal(benchmark_mean_time, goldentime_rocm_e2e, "SDXL e2e benchmark time should not regress")
     mean_line = (f"E2E Benchmark Time: {str(benchmark_mean_time)} ms"
                  f" (golden time {goldentime_rocm_e2e} ms)")
-    job_summary_lines.append(mean_line)
+    e2e_mean_time = benchmark_mean_time
     logging.getLogger().info(mean_line)
 
     # unet benchmark
     ret_value, output = run_sdxl_unet_rocm_benchmark(gpu_number)
     benchmark_mean_time = job_summary_process(ret_value, output)
-    assert benchmark_mean_time <= goldentime_rocm_unet, "SDXL unet benchmark time should not regress"
+    check.less_equal(benchmark_mean_time, goldentime_rocm_unet, "SDXL unet benchmark time should not regress")
     mean_line = (f"Scheduled Unet Benchmark Time: {str(benchmark_mean_time)} ms"
                  f" (golden time {goldentime_rocm_unet} ms)")
-    job_summary_lines.append(mean_line)
+    unet_mean_time = benchmark_mean_time
     logging.getLogger().info(mean_line)
 
+    # unet compilation stats check
+    with open(f"{scheduled_unet_dir}/compilation_info.json", 'r') as file:
+        comp_stats = json.load(file)
+    unet_dispatch_count = int(comp_stats["stream-aggregate"]["execution"]["dispatch-count"])
+    check.less_equal(unet_dispatch_count, goldendispatch_rocm_unet, "SDXL scheduled unet dispatch count should not regress")
```

Comment:
Dispatch count can change and reducing it may improve performance (e.g. enabling more fusions will reduce dispatch count). @MaheshRavishankar what behavior do you want here? 
A) check that the value doesn't change at all, so every PR that affects the dispatch count will need to set these to new values
B) check that the value doesn't change beyond some threshold (e.g. 20%)
C) just log the value without failing the test if it changes from some known value

```
            --goldendispatch-rocm-unet 1718 \
            --goldendispatch-rocm-clip 1571 \
            --goldendispatch-rocm-vae 250 \
```

Reply:
I think what would work is dispatch count shouldnt increase, but it can decrease. I think doing (A) is fine if it is easy to change that value.

Reply:
Sounds good, easy to update, so let's go with checking that the value doesn't change at all to catch any changes in the compilation statistics.